controllers/openstackmachine_controller.go

```diff
@@ -365,6 +365,11 @@ func (r *OpenStackMachineReconciler) reconcileNormal(ctx context.Context, scope
 		return ctrl.Result{}, err
 	}
 
+	result := r.reconcileMachineState(scope, openStackMachine, machine, machineServer)
```

Comment:
given we set the `openStackMachine.Status.Addresses` at line 405, shouldn't we keep the `reconcileMachineState` after that?
If we don't do that I think  we might end up returning the result without the addresses reconciled.